src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/RapidCatchUp.java

```diff
@@ -0,0 +1,242 @@
+package com.hms_networks.americas.sc.extensions.historicaldata;
+
+import com.ewon.ewonitf.Exporter;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPoint;
+import com.hms_networks.americas.sc.extensions.json.JSONException;
+import com.hms_networks.americas.sc.extensions.logging.Logger;
+import com.hms_networks.americas.sc.extensions.string.StringUtils;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUnit;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+/**
+ * This class addresses the need for historical data tracking to "catch up" quickly when there are
+ * periods where the tracker is not running. For example, the Flexy is powered off for several
+ * weeks. For correct behavior, it is required that injectJvmLocalTime from {@link SCTimeUtils} is
+ * called at startup.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.14.2
+ */
+public class RapidCatchUp {
+
+  /**
+   * Threshold for enabling rapid catch up mode in milliseconds. If current time - start time is
+   * less than this amount, catchup mode will not be enabled.
+   */
+  public static final long RAPID_CATCH_UP_TIME_DELTA_MILLISECONDS = SCTimeUnit.HOURS.toMillis(1);
+
+  /** Starting point for request period, which changes based on previous result. */
+  // 10 minutes for the starting point
```

Comment:
```suggestion
  /** 
     * Starting point for request period, which changes based on previous result. 
     * <p>
     * 10 minutes for the starting point.
     * </p>
     */
```

Reply:
im not sure if its really even worth documenting that the value is set to 10 minutes as that is obvious from the code.

Reply:
will change